File: 2.server/src/main/java/ru/vitaliy/petrov/server/forms/requests/documents/AdministrativeOffenceCaseRefusalAddRequest.java
package ru.vitaliy.petrov.server.forms.requests.documents;

import lombok.Data;

import javax.validation.constraints.NotEmpty;
import javax.validation.constraints.NotNull;
import java.sql.Date;
import java.sql.Time;

@Data
public class AdministrativeOffenceCaseRefusalAddRequest {

    @NotNull
    private final Date dateOfFill;

    @NotNull
    private final Time timeOfFill;

    @NotNull
    @NotEmpty
    private final String placeOfFill;

    @NotNull
    private final Long policeOfficerID;

    @NotNull
    @NotEmpty
    private final Long carAccidentInfoSender;

    @NotNull
    @NotEmpty
    private final String carAccidentEstablishedInfo;

    @NotNull
    @NotEmpty
    private final String refusalReason;

    @NotNull
    @NotEmpty
    private final String refusalLawInfo;

    @NotNull
    private final Long carAccidentEntityID;
}

File: client/app/src/main/java/com/petrov/vitaliy/caraccidentapp/domain/models/requests/documents/AdministrativeOffenceCaseRefusalAddRequest.kt
package com.petrov.vitaliy.caraccidentapp.domain.models.requests.documents

import java.sql.Date
import java.sql.Time

data class AdministrativeOffenceCaseRefusalAddRequest(
    val dateOfFill: Date,
    val timeOfFill: Time,
    val placeOfFill: String,
    val policeOfficerID: Long,
    val carAccidentInfoSender: Long,
    val carAccidentEstablishedInfo: String,
    val refusalReason: String,
    val refusalLawInfo: String,
    val carAccidentEntityID: Long
)
